### DISCUSSION TITLE
WebP at Quality = 50 results in 5 times larger file size than JPG

### DISCUSSION THREAD

**Discussion by JwanKhalaf** [SixLabors/ImageSharp#1957] [Show and tell] [answered]
Let me begin by saying **thank you** to all those that are working on this amazing library during their free time. 🙏

I wanted to get your thoughts on something. I am working in a .NET 6 MVC application. I have a form where the user can upload an image file (`.jpg` or `.png` only).

I am using `IFormFile` in my ViewModel, I then turn that into an `Image` from *ImageSharp* so that I can re-size the image, and convert it to `.webp`.

I am using `SixLabors.ImageSharp -Version 2.0.0-alpha.0.159` from [MyGet](https://www.myget.org/feed/sixlabors/package/nuget/SixLabors.ImageSharp). My code "works", but I am surprised by the result. `.webp` is supposed to be a file that is smaller in size than `.jpg`, yet I get:

![image](https://user-images.githubusercontent.com/42877376/150976843-574cbf39-c7cc-4fef-be82-6e6943642e9d.png)

Notice how the `.webp` file is about 5 times larger than the `.jpg` file. I suspect I am doing something silly, I just don't know what.

My whole project is [open-source](https://github.com/JwanKhalaf/Bejebeje.Admin/tree/feat/handle-artist-photo-upload) and I have a branch created for the current work. [Here is where I grab the file](https://github.com/JwanKhalaf/Bejebeje.Admin/blob/19faeba3ff5ce8ece73b0e968dcc9aeaad95de4f/Bejebeje.Services/ArtistService.cs#L241) and turn it into an ImageSharp Image. [Here is where I set the quality of the image to 50](https://github.com/JwanKhalaf/Bejebeje.Admin/blob/19faeba3ff5ce8ece73b0e968dcc9aeaad95de4f/Bejebeje.Services/ArtistImage.cs#L91).

Can anyone see what I'm doing wrong?

**Comment by brianpopow** [comment, marked as answer]
@JwanKhalaf can you attach an example image here, so we can reproduce this?

**Reply by JwanKhalaf** [reply]
Hi @brianpopow. The image is a crappy quality one I was using for testing. But [here it is](https://imgur.com/a/YvzeZUl) 😊

**Reply by brianpopow** [reply]
Ok this explains it. If you do not set the WebP file format when encoding the image, it will default to lossless (we need to make sure to mention this, once we release the a update to the doc's).

If you set it to `Lossy`, the image will be smaller:

```
image.Save("output.webp", new WebpEncoder()
{
   FileFormat = WebpFileFormatType.Lossy
});
```

The output image is 6KB. But keep in mind, that this not guaranteed to be always the case. There can be images which compress better with jpg nonetheless.

**Reply by JimBobSquarePants** [reply]
Is Lossy or Lossless the default with libwebp? This default will cause  a LOT of surprises with ImageSharp.Web.

**Reply by brianpopow** [reply]
I have just checked `libwebp`: Lossy is the default there. I think we should should change ours to match that. 

edit: changed with #1961

**Reply by JimBobSquarePants** [reply]
Yep definitely. 👍

**Reply by JwanKhalaf** [reply]
Thank you so much @brianpopow! You were right! After making the change, it now works as expected.

![image](https://user-images.githubusercontent.com/42877376/151243271-e2395aec-21c9-4528-bcfb-241a3a45364f.png)

**Comment by br3aker** [comment]
> `.webp` is supposed to be a file that is smaller in size than `.jpg`

It **really depends** on the input image.
